File: s1/s1/src/main/java/com/nodal/s1/publisher/RabbitMQPublisher.java
package com.nodal.s1.publisher;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.amqp.rabbit.core.RabbitTemplate;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

@Service
public class RabbitMQPublisher {

    private static final Logger LOGGER = LoggerFactory.getLogger(RabbitMQPublisher.class);

    @Value("${rabbitmq.queue.s2.send}")
    private String sendQueue;

    @Value("${rabbitmq.exchange.name}")
    private String exchange;

    @Value("${rabbitmq.routing.key.s2.send}")
    private String sendS2RoutingKey;

    private final RabbitTemplate rabbitTemplate;

    @Autowired
    public RabbitMQPublisher(RabbitTemplate rabbitTemplate) {
        this.rabbitTemplate = rabbitTemplate;
    }

    public void sendMessage(String message){
        LOGGER.info("Message Sent by S1 to rqm: {}", message);
        rabbitTemplate.convertAndSend(exchange,sendS2RoutingKey,message);
    }

}

File: s2/s2/src/main/kotlin/com/nodal/s2/publisher/RabbitMQPublisher.kt
package com.nodal.s2.publisher

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.amqp.rabbit.annotation.RabbitListener
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class RabbitMQPublisher {

    val LOGGER: Logger = LoggerFactory.getLogger(RabbitMQPublisher::class.java)

    @Value("\${rabbitmq.queue.s1.send}")
    private val sendQueue: String? = null

    @Value("\${rabbitmq.exchange.name}")
    private val exchange: String? = null

    @Value("\${rabbitmq.routing.key.s1.send}")
    private val sendS1RoutingKey: String? = null

    private var rabbitTemplate: RabbitTemplate? = null

    @Autowired
    fun RabbitMQPublisher(rabbitTemplate: RabbitTemplate) {
        this.rabbitTemplate = rabbitTemplate
    }

    fun sendMessage(message: String) {
        LOGGER.info("Message Sent by S2: {}", message)
        //!! -> not null assertion
        rabbitTemplate!!.convertAndSend(exchange!!, sendS1RoutingKey!!, message)
    }
}